### DISCUSSION TITLE
Ruffle in Microsoft Office as an add-in or ActiveX control

### DISCUSSION THREAD

**Discussion by nphuracm** [ruffle-rs/ruffle#14244] [Ideas]
So in my area, I have seen some teachers still employing Flash in their PowerPoint presentations for class. Since Flash is gone, the Flash animations all broke. So I wonder if it is possible or feasible to integrate Ruffle in PowerPoint (or other office suites like LibreOffice?) to make the Flash anims work again.

**Comment by n0samu** [comment]
Hi, here is a tutorial to extract Flash animations from your PowerPoints: https://www.youtube.com/watch?v=Uib84NRm-GY
I think it should be fairly simple to create a modern Office Add-In that allows inserting Flash animations into PowerPoints and playing them with the web version of Ruffle. This would still require extracting and reinserting the Flash content, though.
